File: discover_ruby/manipulate_the_data.rb
require 'HTTPClient'
require 'json'

extheaders = {
  'User-Agent' => 'Holberton_School',
  'Authorization' => 'token 914b7070c282e25764284a8375e636a4302ac5d7'
}
# create simple client
http = HTTPClient.new
# retrieve content from specified URL
request = http.get_content('https://api.github.com/search/repositories?q=language:ruby&sort=stars&order=desc')
# parse json response
json = JSON.parse(request)
# iterate through parsed json outputting full names
puts json["items"].map {|i| i["full_name"]}.join("\n")

File: discover_more_python/manipulate_the_data.py
import requests
import json
# function to parse json and print specific field
request_headers = {
  'User-Agent': 'Holberton_School',
  'Authorization': 'token b28c7b657c6fd8ba06b3d8bb7bf1835d1f50b40f'
}

r = requests.get('https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc', request_headers)
c = r.content
j = json.loads(c)

for k in j['items']:
		print k['full_name']
